kasa/smart/smartdevice.py

```diff
@@ -486,3 +486,26 @@ async def factory_reset(self) -> None:
         Note, this does not downgrade the firmware.
         """
         await self.protocol.query("device_reset")
+
+    @property
+    def device_type(self) -> DeviceType:
+        """Return the device type."""
+        if self._device_type is not DeviceType.Unknown:
+            return self._device_type
```

Comment:
We are not initing this before someone accesses this property -> calling `is_*` doesn't work currently. Perhaps this should be done automatically in the initial update?

Reply:
I think it works ok being in the property itself as it makes it easy to find the logic.  Why doesn't calling `is_*` work?

Reply:
Ah, looks like we need to adjust `is_*` to use the property, and not the internal presentation. At the moment `Device.is_bulb` returns `self._device_type` so that's why my previous tests that were checking those properties was not working.